src/modules/class-wpml-cornerstone-module-with-items.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+/**
+ * Class WPML_Cornerstone_Module_With_Items
+ */
+abstract class WPML_Cornerstone_Module_With_Items implements IWPML_Page_Builders_Module {
```

Comment:
@adrianosferreira It's a pity this is virtually the same as `WPML_Beaver_Builder_Module_With_Items` and `WPML_Elementor_Module_With_Items` with some minor variations. 

Can you add a separate task to the cornerstone work to see if they can be integrated so the code can be reused.

Reply:
@brucepearson these minor variations are not so minor. Besides the abstract methods definition, the other methods like `get`, `update` and `get_items` have a very specific implementation which depends on the format of custom field:

- `get` and `update`: looking at BB you can see that the custom field is an object while Elementor and Cornerstone are arrays. So, locating the field value of a node will be different for all libraries

- `get_items`: again, these three libraries store `items` at different places. BB `$settings->items`, Elementor `$element[ WPML_Elementor_Translatable_Nodes::SETTINGS_FIELD ][ $this->get_items_field() ]`, and Cornerstone ` $settings[ $this->get_items_field() ]`

That's why I think each lib should have its own class. Unfortunately, they don't follow a common pattern for structuring the custom field which makes hard for us to create something common for repeater module fields.